src/utils/guardToPromiseFn.ts

```diff
@@ -19,9 +19,15 @@ export function guardToPromiseFn(
         valid?: boolean | RouteLocation
       ) => {
         // TODO: handle callback
-        if (valid === false) reject(new NavigationAborted(to, from))
+        if (valid === false)
+          reject(createRouterError(ErrorTypes.NAVIGATION_ABORTED, { from, to }))
         else if (isRouteLocation(valid)) {
-          reject(new NavigationGuardRedirect(to, valid))
+          reject(
+            createRouterError(ErrorTypes.NAVIGATION_GUARD_REDIRECT, {
+              from: to,
+              to: valid as RouteLocationNormalized, // todo: remove as
```

Comment:
here `valid` is just `RouteLocation`, how should we make it normalized?

Reply:
We don't need to make it normalised for the redirect. In that case the `to` property can either be `RouteLocation` or `RouteLocationNormalized`. I forgot about it in my previous review 😅